processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -375,14 +375,16 @@ public Type erasure() {
      *
      * @return {@code true} if and only if this type is assignable to the given other type.
      */
-    // TODO This doesn't yet take wild card types into account; e.g. ? extends Integer wouldn't be assignable to Number
-    // atm.
     public boolean isAssignableTo(Type other) {
         if ( equals( other ) ) {
             return true;
         }
 
-        return typeUtils.isAssignable( typeMirror, other.typeMirror );
+        TypeMirror typeMirrotToMatch = isWildCardExtendsBound() ? getTypeBound().typeMirror : typeMirror;
+        TypeMirror otherTypeMirrotToMatch = other.isWildCardSuperBound() ?
```

Comment:
I do not have any practicle use case for this scenario. I think it is right but anyhow if the second type is `? super AnyType` it resolves to `Object` and hence any type would be assignable to this? Very theoretically. Maybe it would be better to leave the super-wildcard-case and only implement the extends-wildcard-case.
Further more if there would be a `List<? super AnyType>` (for which I do not have any practical use case) any iteration over that list must use Object and hence our Type#getTypeBound would lead to the wrong type.